keydata_test.go

```diff
@@ -1322,6 +1256,7 @@ func (s *keyDataSuite) TestReadAndWriteWithUnsaltedKeyDigest(c *C) {
 	auxKey := testutil.DecodeHexString(c, "8107f1c65c58934f0d59245d1d94d312ea803e69c8599a7bac8c67fe253232f2")
 	j := []byte(
 		`{` +
+			`"version":0,` +
```

Comment:
As existing keys won't have this field, we should test that reading works for payloads that don't contain it.

Reply:
Correct, I got confused here. So I modified the version behavior to use 0 of 1 for the version field if the `Version` is less than 2 and by marking the Version field as omitempty but I am not sure if that is correct. Can you think of any other legacy scenario that might be weird here?

I also had to modify the tests in `keydata_legacy_test.go` to use 0 for legacy version. Is keyData with version 1 possible?

Reply:
I added a comment separately for this - I think the `omitempty` is fine, but I would still fake version 1 in `KeyData.Version` so that it returns 1 if the field is 0. I'm not sure if it matters either way to be honest, but it's a bit weird going from v0 -> v2.